languages/rust/reference/implementing-a-concept-exercise.md

```diff
@@ -1,5 +1,75 @@
-# How to implement an Rust concept exercise
+# How to implement a Rust concept exercise
 
-TODO: describe how to implement a concept exercise for the Rust track. For inspiration, check out the [C# version of this file][csharp-implementing].
+This issue describes how to implement a concept exercise for the Rust track. As this document is generic, the following placeholders are used:
 
-[csharp-implementing]: ../../csharp/reference/implementing-a-concept-exercise.md
+- `<SLUG>`: the name of the exercise in kebab-case (e.g. `anonymous-methods`).
+- `<NAME>`: the name of the exercise in snake_case (e.g. `anonymous_methods`).
+
+Before implementing the exercise, please make sure you have a good understanding of what the exercise should be teaching (and what not). This information can be found in the exercise's GitHub issue. Having done this, please read the [Rust concept exercises introduction][concept-exercises].
+
+To implement a Rust concept exercise, the following files must be created:
+
+```
+languages
+└── rust
+    └── exercises
+        └── concept
+            └── <SLUG>
+                ├── .docs
+                |   ├── instructions.md
+                |   ├── introduction.md
+                |   ├── hints.md
+                |   └── after.md (optional)
+                ├── .meta
+                |   |── design.md
+                |   |── config.json
+                |   └── example.rs
+                ├── src
+                |   └── lib.rs
+                ├── Cargo.toml
+                └── tests
+                    └── <NAME>.rs
+```
+
+## Step 1: adding track-specific files
+
+These files are specific to the Rust track:
```

Comment:
We may want to add a line about how to use `cargo new my-concept-exercise` to bootstrap the initial concept exercise.

Reply:
Has to be `cargo (init|new) --lib concept-exercise` because exercism projects are tapped as libraries, but yes, that can be done.

Actually, a complete `cargo-exercise.sh` script can be included, I only hesitate because I don't want to be exclusionary to Windows users if it doesn't work for them and frankly no, I'm not entirely sure if a POSIX-compatible `.sh` works on most Windows dev environments.